src/main/java/net/openhft/chronicle/wire/BinaryWire.java

```diff
@@ -460,7 +473,13 @@ public void readWithLength(@NotNull WireOut wire, int len) throws InvalidMarshal
         try {
             bytes.readLimit(newLimit);
             @NotNull final ValueOut wireValueOut = wire.getValueOut();
-            switch (getBracketTypeNext()) {
+            BracketType bracketType = getBracketTypeNext();
+            switch (bracketType) {
+                case UNKNOWN:
+                    bytes.uncheckedReadSkipOne();
+                    copyHistoryMessage(wire);
+                    return;
```

Comment:
Is it right to use `UNKNOWN` to convey `HISTORY_MESSAGE` here? seems a bit obfuscated to me.